src/core/service.c

```diff
@@ -2578,16 +2580,18 @@ static void service_notify_cgroup_empty_event(Unit *u) {
 
         case SERVICE_START:
         case SERVICE_START_POST:
-                /* If we were hoping for the daemon to write its PID file,
-                 * we can give up now. */
-                if (s->pid_file_pathspec) {
+                if (s->type == SERVICE_NOTIFY)
+                        /* No chance of getting a ready notification anymore */
+                        service_enter_signal(s, SERVICE_FINAL_SIGTERM, SERVICE_FAILURE_PROTOCOL);
```

Comment:
So notifications can come from outside the cgroup? That is nasty. Is this really desirable? If so, we should keep track of processes that are allowed to send notifications and enter FAILURE_PROTOCOL when none are left...

Shouldn't we put the main_pid / control_pid in the cgroup created as part of the PAM session? If we already do (could be, I haven't checked) then the cgroup_empty event is called erroneously.